javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -22,6 +22,78 @@
  */
 final class Collections {
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    static boolean equals(Map<?, ?> map1, Object object) {
+        if (map1 == null ^ object == null) {
```

Comment:
Yes, until it is used only in equals, but we shouldn't forget how to use this static method. Maybe annotate source with ```@NotNull```?